src/commands/DriveToWaypointCommand.cpp

```diff
@@ -0,0 +1,56 @@
+#include "DriveToWaypointCommand.h"
+#include "../log.h"
+
+using namespace navtypes;
+using namespace robot::types;
+using namespace std::chrono_literals;
+
+static std::chrono::duration CLOSE_TO_TARGET_DUR = 1s;
+
+namespace commands {
+
+DriveToWaypointCommand::DriveToWaypointCommand(const point_t& target, double thetaKP,
+											   double driveVel, double slowDriveVel,
+											   double doneThresh)
+	: target(target), pose(pose_t::Zero()), thetaKP(thetaKP), driveVel(driveVel),
+	  slowDriveVel(slowDriveVel), doneThresh(doneThresh), setStateCalledBeforeOutput(false) {}
+
+void DriveToWaypointCommand::setState(const navtypes::pose_t& pose) {
```

Comment:
To make testing and usage easier, we probably shouldn't call the clock for the current time, let's take it as an argument here and store it as a field.